src/coreclr/pal/src/thread/process.cpp

```diff
@@ -2524,7 +2524,7 @@ PROCAbort(int signal, siginfo_t* siginfo)
 
     // Restore all signals; the SIGABORT handler to prevent recursion and
     // the others to prevent multiple core dumps from being generated.
-    SEHCleanupSignals();
+    SEHCleanupSignals(false /* isChildProcess */);
```

Comment:
How high is the risk of reentrancy here? We probably don't want to end in a recursive case here.

Reply:
ProcAbort is calling abort() right after this line, so I don't see how it could be reentered on the same thread.